### PR TITLE
OEL-1450: EPIC-List-pages.

### PR DESCRIPTION
**Pull request by drishu**
### Description

Integrate oe_list_pages, provide facets for news and event, all packaged under a new oe_showcase_list_pages module.

### Change log

- Added: oe_list_pages, factes
- Changed: permissions for editor role

### CODE REVIEW

File: modules/oe_showcase_list_pages/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# OpenEuropa Showcase List Pages
+
+The scope of this module is to demo the OpenEuropa Library List pages.
+This module uses [Open Europa Listing Page](https://github.com/openeuropa/oe_list_pages) and enables [Open Europa Whitelabel List Pages](https://github.com/openeuropa/oe_whitelabel). This module contains some facets exported and a dedicated index
+and server for the search.
+
+## Create a list page and setting up bundles and filters
+By default this module enables oe_list_pages module, the "news" bundle as well as the fields "title" and "publication date" filters.
```
Comment (drishu) [inline modules/oe_showcase_list_pages/README.md:8]: this sentence is not ok, it does not enable the "news" bundle

File: modules/oe_showcase_list_pages/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+### To set up new bundles
+
+* Go to `config/search/search-api`.
+* Add new index.
+* Check Content on "Datasources".
+* Select the desired bundle(s).
+* Click on Save.
+* Click on the tab fields to add the fields do be indexed.
+* Click on save.
+
```
Comment (drishu) [inline modules/oe_showcase_list_pages/README.md:26]: Lets make a special mention that if more then one bundle is indexed then we need the content type field indexed

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    $search_button->click();
+    $this->assertSearchResultsTitle('Event list page', 1);
+    $this->assertSearchResults([
+      'Event number 8',
+    ]);
+
+    // Assert only Event nodes are part of the result.
+    $title_input->setValue('News number 1');
+    $search_button->click();
+    $this->assertSearchResultsTitle('Event list page', 0);
```
Comment (drishu) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php:266]: this test is la lacking more of who can do what, we should use editor only when setting up content, but use anonymous when using the listing page.
Also some of these methods could use better naming.

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+
+    // Create editor user.
+    $user = $this->createUser([]);
+    $user->addRole('editor');
+    $user->save();
+    $this->drupalLogin($user);
+  }
+
+  /**
+   * Check creation List pages content through the UI.
```
Comment (drishu) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php]: could use proper English

File: modules/oe_showcase_list_pages/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+### Set up new filters
+Filters are facets on search API module. To create new facets you should:
+* Go to `admin/config/search/facets`.
+* Click on "+Add facet".
```
Comment (drishu) [inline modules/oe_showcase_list_pages/README.md:30]: we need to mention which widget to use for which field type

File: oe_showcase.post_update.php [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+ */
+
+declare(strict_types = 1);
+
+/**
+ * Implements hook_post_update_NAME().
+ *
+ * Install "OpenEuropa showcase listing pages" module.
+ */
+function oe_showcase_post_update_10001(&$sandbox): void {
```
Comment (drishu) [inline oe_showcase.post_update.php]: just a remark, lets no forget to adapt this!

File: oe_showcase.post_update.php [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * @file
+ * OpenEuropa showcase post updates.
+ */
+
+declare(strict_types = 1);
+
+/**
+ * Implements hook_post_update_NAME().
```
Comment (drishu) [inline oe_showcase.post_update.php]: remove this line

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php (hunk 3 above)
Comment (drishu) [reply modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php:266]: this one we can follow up some other time

**Review by brummbar — APPROVED** [review]

**Review by escuriola — APPROVED** [review]

**Pull request merged**